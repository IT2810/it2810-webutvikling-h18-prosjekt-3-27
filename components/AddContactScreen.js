import React, { Component } from "react";
import {
  StyleSheet,
  View,
  TextInput,
  Button
} from "react-native";

export default class AddContactScreen extends Component {
  state = {
    name: "",
    number: "",
  };

  changeTextHandlerName = name => {
    this.setState({name: name});
  };

  changeTextHandlerNumber = number => {
    this.setState({number: number});
  };

  handleAddPress = () => {
    if (this.state.name.trim().length === 0) {
      // invalid name
      return;
    }
    // retrieve addContact(contact) callback function
    const addContact = this.props.navigation.getParam("addContact");
    // create contact
    const newContact = {
      name: this.state.name.trim(),
      number: this.state.number.trim()
    };
    addContact(newContact);
    this.props.navigation.navigate("Contacts");
  };

  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.input}
                   onChangeText={this.changeTextHandlerName}
                   value={this.state.name}
                   placeholder="Name"
        />
        <TextInput style={styles.input}
                   onChangeText={this.changeTextHandlerNumber}
                   value={this.state.number}
                   placeholder="Phone"
        />
        <Button
          color='pink'
          onPress={this.handleAddPress}
          value={this.state.name}
          title={"Add"}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "stretch",
    backgroundColor: "white",
  },
  input: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    margin: 20,
  }
});

