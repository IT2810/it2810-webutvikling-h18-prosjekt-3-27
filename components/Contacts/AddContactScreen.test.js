import React from "react";
import AddContactScreen from './AddContactScreen';


//Simple snapshot test for addcontactscreen
test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<AddContactScreen/>);
  expect(tree).toMatchSnapshot();
});