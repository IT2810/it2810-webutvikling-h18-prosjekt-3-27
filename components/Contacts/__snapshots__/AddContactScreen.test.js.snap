// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Name"
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 20,
        "padding": 5,
      }
    }
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Phone"
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 20,
        "padding": 5,
      }
    }
    value=""
  />
  <Button
    color="green"
    onPress={[Function]}
    title="Add"
    value=""
  />
</Component>
`;
